source/s3_meta_request.c

```diff
@@ -114,9 +119,27 @@ static void s_s3_request_on_body(
     if (aws_py_gilstate_ensure(&state)) {
         return; /* Python has shut down. Nothing matters anymore, but don't crash */
     }
-
-    PyObject *result = PyObject_CallMethod(
-        request_binding->self_py, "_on_body", "(y#K)", (const char *)(body->ptr), (Py_ssize_t)body->len, range_start);
+    PyObject *result = NULL;
+    if (request_binding->file_py) {
+        if (body->len) {
+            PyObject *byte_object = PyBytes_FromStringAndSize((char *)(body->ptr), (Py_ssize_t)body->len);
+            if (PyFile_WriteObject(byte_object, request_binding->file_py, 0)) {
+                /* TODO: raise error */
+                goto done;
+            }
```

Comment:
The byte object will be converted to a string representation, https://github.com/python/cpython/blob/master/Objects/fileobject.c#L135. We need to come up with another approach if we want it to work well...
I guess we will have to create a FILE * from the fileno() given by Python in C land. Or @JonathanHenson, do you have any suggestion?